src/windows/wslaservice/exe/WSLAContainer.cpp

```diff
@@ -196,35 +197,93 @@ CATCH_RETURN();
 HRESULT WSLAContainer::Exec(const WSLA_PROCESS_OPTIONS* Options, IWSLAProcess** Process, int* Errno)
 try
 {
-    // auto process = wil::MakeOrThrow<WSLAProcess>();
+    *Errno = -1;
+    THROW_HR_IF_MSG(E_INVALIDARG, Options->Executable != nullptr, "Executable must be null");
+
+    std::lock_guard lock{m_lock};
```

Comment:
Is this the C++17 thing of auto-deducing template args? Not seeing the `std::lock_guard<std::recursive_mutex> lock{m_lock};` here